type-to-search/index.js

```diff
@@ -1,36 +1,45 @@
 import { input } from '@most/dom-event'
-import { map, filter, debounce, skipRepeats, switchLatest, fromPromise } from 'most'
+import { map, filter, debounce, skipRepeats, switchLatest, fromPromise, merge } from 'most'
 import rest from 'rest/client/jsonp'
 
-  const url = 'https://en.wikipedia.org/w/api.php?action=opensearch&format=json&search='
+const url = 'https://en.wikipedia.org/w/api.php?action=opensearch&format=json&search='
 
-  const search = document.getElementById('search')
-  const resultList = document.getElementById('results')
-  const template = document.getElementById('template').innerHTML
+const search = document.getElementById('search')
+const resultList = document.getElementById('results')
+const template = document.getElementById('template').innerHTML
 
-  // Fetch results with rest.js
-  // Returns a promise for the wikipedia json response
-  const getResults = text => rest(url + text).entity()
+// Fetch results with rest.js
+// Returns a promise for the wikipedia json response
+const getResults = text => rest(url + text).entity()
 
-  // Debounce keystrokes and get input value when it changes
-  // Only search if the user stopped typing for 500ms, and if the
-  // text is longer than 1 character and is different than the last
-  // time we saw the text.
-  const searchText = input(search)
-    .debounce(500)
-    .map(e => e.target.value.trim())
-    .filter(text => text.length > 1)
-    .skipRepeats()
+// Get input value when it changes
+const searchText = input(search)
+  .map(e => e.target.value.trim())
+  .skipRepeats()
```

Comment:
It's still a bit more efficient to multicast here, since merge() will observe both `results` and `emptyResults`.  It's harmless, though, to omit it in this case ... the only real downside is that the function passed to `map` above will be invoked twice for each `input` event in the search box.

Let's re-add it, and I think this is all ready to go!